var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -90,14 +95,17 @@ class PyScipy(PythonPackage):
     depends_on("py-setuptools@:59", when="@1.8", type="build")
     depends_on("py-setuptools@:57", when="@1.7", type="build")
     depends_on("py-setuptools@:51.0.0", when="@1.6", type="build")
-    depends_on("py-numpy@1.21.6:1.27", when="@1.11:", type=("build", "link", "run"))
+    depends_on("py-numpy@1.22.4:1.28", when="@1.12:", type=("build", "link", "run"))
+    depends_on("py-numpy@1.21.6:1.27", when="@1.11", type=("build", "link", "run"))
     depends_on("py-numpy@1.19.5:1.26", when="@1.10", type=("build", "link", "run"))
     depends_on("py-numpy@1.18.5:1.25", when="@1.9", type=("build", "link", "run"))
     depends_on("py-numpy@1.17.3:1.24", when="@1.8", type=("build", "link", "run"))
     depends_on("py-numpy@1.16.5:1.22", when="@1.6:1.7", type=("build", "link", "run"))
     depends_on("py-numpy@1.14.5:1.21", when="@1.5", type=("build", "link", "run"))
     depends_on("py-numpy@1.13.3:1.21", when="@1.3:1.4", type=("build", "link", "run"))
     depends_on("py-pytest", type="test")
+    depends_on("py-pooch", type="test")
+    depends_on("py-hypothesis@6.30:", type="test")
```

Comment:
Not documented in pyproject.toml, but I needed to add a newer version of hypothesis to get things working. For older versions of hypothesis, I got the following error:
```
TypeError: from_dtype() got an unexpected keyword argument 'allow_subnormal'
```
Might be worth documenting in pyproject.toml.

Reply:
https://github.com/scipy/scipy/pull/19942